includes/discovery/sensors/state/equallogic.inc.php

```diff
@@ -141,3 +141,102 @@
+    .1.3.6.1.4.1.12740.3.1.1.1.5.1.329840783.13 = RZ1wTmZ00983
+    .1.3.6.1.4.1.12740.3.1.1.1.5.1.329840783.14 = RZ1wTmZ0096D
+    .1.3.6.1.4.1.12740.3.1.1.1.5.1.329840783.15 = RZ1wTmZ00952
+    .1.3.6.1.4.1.12740.3.1.1.1.5.1.329840783.16 = RZ1wTmZ00962
+    .1.3.6.1.4.1.12740.3.1.1.1.5.1.329840783.17 = RZ1wTmZ0095N
+    .1.3.6.1.4.1.12740.3.1.1.1.5.1.329840783.18 = RZ1wTmZ0097M
+    .1.3.6.1.4.1.12740.3.1.1.1.5.1.329840783.19 = 6RZ1wTmZ3QS5
+    .1.3.6.1.4.1.12740.3.1.1.1.5.1.329840783.20 = RZ1wTmZ0091H
+    .1.3.6.1.4.1.12740.3.1.1.1.5.1.329840783.21 = RZ1wTmZ00981
+    .1.3.6.1.4.1.12740.3.1.1.1.5.1.329840783.22 = RZ1wTmZ0097H
+    .1.3.6.1.4.1.12740.3.1.1.1.5.1.329840783.23 = RZ1wTmZ00959
+    .1.3.6.1.4.1.12740.3.1.1.1.5.1.329840783.24 = RZ1wTmZ0098B
+**/
+
+$base_oid         = '.1.3.6.1.4.1.12740.3.1.1.1.8.1.'; // eqlDiskStatus
+
+if (!empty($oids_disks)) {
+/*
+eqlDiskStatus
+    INTEGER {
+            on-line  (1),
+            spare    (2),
+            failed   (3),
+            off-line (4),
+            alt-sig  (5),
+            too-small(6),
+            history-of-failures(7),
+            unsupported-version(8)
+    }
+*/
+    $state_name1 = 'eqlDiskStatus';
+    $state_index_id1 = create_state_index($state_name1);
+
+    if ($state_index_id1) {
+        $states1 = array(
+            array($state_index_id1,'on-line',1,1,0),
+            array($state_index_id1,'spare',1,2,0),
+            array($state_index_id1,'failed',1,3,2),
+            array($state_index_id1,'off-line',1,4,1),
+            array($state_index_id1,'alt-sig',1,5,1),
+            array($state_index_id1,'too-small',1,6,2),
+            array($state_index_id1,'history-of-failures',1,7,1),
+            array($state_index_id1,'unsupported-version',1,8,1),
+        );
+        foreach ($states1 as $value) {
+            $insert = array(
+                'state_index_id' => $value[0],
+                'state_descr' => $value[1],
+                'state_draw_graph' => $value[2],
+                'state_value' => $value[3],
+                'state_generic_value' => $value[4]
+            );
+            dbInsert($insert, 'state_translations');
+        }
+    }
+
+    foreach (explode("\n", $oids_disks) as $data) {
+        $data = trim($data);
+        if (!empty($data)) {
+            list($oid,$descr) = explode(' = ', $data, 2);
+            $split_oid        = explode('.', $oid);
+            $disk_index        = $split_oid[(count($split_oid) - 1)];
+            $index            = (int)$disk_index+0;
+            $member_id        = $split_oid[(count($split_oid) - 2)];
+            $num_index        = $member_id.'.'.$disk_index;
+            $oid              = $base_oid.$num_index;
+            $extra            = snmp_get_multi($device, $oid, '-OQne', 'EQLDISK-MIB', 'equallogic');
+            d_echo($extra);
+            if (!empty($extra)) {
+                list($foid,$pstatus) = explode(' = ', $extra, 2);
+                $index        = (100 + $index);
```

Comment:
It's probably better to use eqlDiskStatus.'.'.$disk_index here (need to also update create_sensor_to_state() as well. It will just stop the hack with adding 100 to the index).